### PR TITLE
Documentation Cleanup

### PR DESCRIPTION
**Pull request by RDIL**
## PR Summary
- Fixed MDLint Issues
- Fixed Grammar Issues
- Added/fixed links

Also sorry I had to reuse an old patch branch meaning one of Dongbo's commits is still there

  

## PR Context  

Improves parts of documentation.  
## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [x] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [X] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

File: docs/dev-process/coding-guidelines.md [hunk 1]
```diff
@@ -111,8 +111,7 @@ Some general guidelines:
 * Avoid creating empty arrays.
   Instead, reuse the static ones via `Utils.EmptyArray<T>`.
 
-* Avoid unnecessary memory allocation in a loop.
-  Move the memory allocation outside the loop if possible.
+* Avoid unnecessary memory allocation in a loop. Move the memory allocation outside the loop if possible.
```
Comment (TravisEz13) [inline docs/dev-process/coding-guidelines.md]:
```suggestion
* Avoid unnecessary memory allocation in a loop.
  Move the memory allocation outside the loop if possible.
```

A new sentence should always be on the next line

File: docs/maintainers/issue-management.md [hunk 2]
```diff
@@ -53,15 +53,15 @@ These labels describe what feature area of PowerShell that an issue affects:
...
 * `Area-Intellisense`: tab completion
 * `Area-Language`: parser, language semantics
-* `Area-OMI`: omi
+* `Area-OMI`: OMI
 * `Area-PackageManagement`: PackageManagement related issues
 * `Area-Performance`: a performance issue
 * `Area-Portability`: anything affecting script portability
 * `Area-PowerShellGet`: PowerShellGet related issues
-* `Area-Providers`: PowerShell providers like FileSystem, Certificates, Registry, etc...
+* `Area-Providers`: PowerShell providers such as FileSystem, Certificates, Registry, etc...
```
Comment (TravisEz13) [inline docs/maintainers/issue-management.md]: @sdwheeler 	Do you have any comments on which is more appropriate here?

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: docs/maintainers/issue-management.md (hunk 2 above)
Comment (RDIL) [reply docs/maintainers/issue-management.md]: `like` is not very professional sounding.  `such as` just sounds better in my opinion.

File: docs/maintainers/issue-management.md [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 If you believe that there is a security vulnerability in PowerShell Core,
 it **must** be reported to [secure@microsoft.com](https://technet.microsoft.com/security/ff852094.aspx) to allow for [Coordinated Vulnerability Disclosure](https://technet.microsoft.com/security/dn467923).
-**Only** file an issue, if secure@microsoft.com has confirmed filing an issue is appropriate.
+**Only** file an issue, if [secure@microsoft.com][https://technet.microsoft.com/security/ff852094.aspx] has confirmed filing an issue is appropriate.
```
Comment (sdwheeler) [inline docs/maintainers/issue-management.md]: The URL should be https://www.microsoft.com/en-us/msrc/faqs-report-an-issue?rtc=1

File: docs/maintainers/issue-management.md (hunk 2 above)
Comment (sdwheeler) [reply docs/maintainers/issue-management.md]: Either way is fine. No preference.

**Review by sdwheeler — CHANGES_REQUESTED** [review]

File: docs/maintainers/issue-management.md (hunk 3 above)
Comment (RDIL) [reply docs/maintainers/issue-management.md]: Fixed.

**Review by TravisEz13 — APPROVED** [review]

**Comment by RDIL** [comment]
@sdwheeler please update your review

**Review by sdwheeler — APPROVED** [review]

File: docs/building/internals.md [hunk 4]
```diff
@@ -143,7 +143,7 @@ Be sure to build and test for all supported architectures: `x86`, `x64`, `x64_ar
 The `x64_arm` and `x64_arm64` architectures mean that the host system needs to be x64 to cross-compile to ARM.
 When building for multiple architectures, be sure to use the `-clean` switch as cmake will cache the previous run and the wrong compiler will be used to generate the subsequent architectures.
 
-After that, the binary `pwrshplugin.dll`, its PDB file, and `powershell.core.instrumentation.dll` will be placed under 'src\powershell-win-core'.
+After that, the binary `pwrshplugin.dll`, its PDB file, and `powershell.core.instrumentation.dll` will be placed under `src\powershell-win-core`.
```
[the remark this answers is not in the document]
Comment (RDIL) [reply docs/building/internals.md]: Perhaps so.

**Review by TravisEz13 — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

File: docs/dev-process/breaking-change-contract.md [hunk 5]
```diff
@@ -4,29 +4,33 @@ We have a serious commitment to backwards compatibility with all earlier version
 
 Below is a summary of our approach to handing breaking changes including what kinds of things constitute breaking changes, how we categorize them, and how we decide what we're willing to take.
 
-Note that these rules only apply to existing stable features that have shipped in a supported release. New features marked as “in preview” that are still under development may be modified from one preview release to the next. These are not considered breaking changes.
+Note that these rules only apply to existing stable features that have shipped in a supported release. New features marked as “in preview” that are still under development may be modified from one preview release to the next.
```
[the remark this answers is not in the document]
Comment (RDIL) [reply docs/dev-process/breaking-change-contract.md]: agreed

**Comment by iSazonov** [comment]
@TravisEz13 Could you merge?

**Comment by RDIL** [comment]
Commenting to make sure stale bot doesn't tag the PR.  
@TravisEz13 this should be ready for merge.

**Pull request merged**